web/src/filter.ts

```diff
@@ -207,6 +207,17 @@ function message_matches_search_term(message: Message, operator: string, operand
             return message.stream_id.toString() === operand;
         }
 
+        case "channels":
+            if (message.type !== "stream") {
+                return false;
+            }
+            switch (operand) {
+                case "archived":
+                    return stream_data.is_stream_archived(message.stream_id);
+                default:
+                    return false;
```

Comment:
This remains wrong. You should handle all valid `operand` values, and throw an exception for unexpected/unsupported values here, rather than giving a wrong answer.

Reply:
`channels:web-public` and `channels:public` are I think the others that exist.

Reply:
`invite_only` is the API field that corresponds to public-or-not.

We don't need to support something that doesn't exist yet. But your code is wrong and will make `channels:public` views not work correctly.